R/recompose.R

```diff
@@ -1,50 +1,88 @@
-# This is the same as the "recomposition" at the end of recipes::bake()
-
-recompose <- function(data, composition) {
-  if (identical(composition, "tibble")) {
-    data
-  } else if (identical(composition, "dgCMatrix")) {
-    convert_matrix(data, sparse = TRUE)
-  } else if (identical(composition, "matrix")) {
-    convert_matrix(data, sparse = FALSE)
-  } else {
-    abort("Internal error: Unknown `composition` type.")
-  }
-}
+#' Recompose a data frame into another form
+#'
+#' @description
+#' `recompose()` takes a data frame and converts it into one of:
+#' - A tibble
+#' - A data frame
```

Comment:
I added support for `"data.frame"` specifically for recipes, but hardhat won't use it